### ISSUE TITLE
all: failures during bootstrapping on `linux-riscv64-unmatched`

### ISSUE THREAD

**Issue by mknyszek** [golang/go#62104]
```
#!watchflakes
post <- (pkg ~ `^bootstrap` || log ~ `^bootstrap/`) && builder == "linux-riscv64-unmatched" && test == ""
```

2023-08-08 21:48 linux-riscv64-unmatched go@6e434079 bootstrap/cmd/link/internal/ld [build] ([log](https://build.golang.org/log/56a00d7cdf8fd3a7f8a9991efbb7eb491a10acb1))
2023-08-09 15:15 linux-riscv64-unmatched go@cd589c8a bootstrap/debug/dwarf [build] ([log](https://build.golang.org/log/b1c9ebad3ed091f7c1a1800ec968f88edd84d9b7))

CC @golang/riscv64

**Comment by mengzhuo** [comment]
@bcmills  Can we upgrade GOROOT_BOOTSTRAP to Go 1.20.7? This fixed a LD/SC asm code generation issue, I think this might related.

**Comment by bcmills** [comment]
@mengzhuo, I'm not sure exactly what the process for that is. 😅

It looks like @cagedmantis will be on the @golang/release interrupts rotation on Monday — perhaps he can help with that?

**Comment by dmitshur** [comment]
I might be able to help a bit sooner too. :)

Sent [CL 520900](https://go.dev/cl/520900).

FWIW, if you think it has a chance to help more, it's also fine to use go1.21.0 as the bootstrap for this builder. As [cmd/dist/README](https://go.googlesource.com/go/+/a9859a7df72a8ca9686b4fb939989c04b42804fe/src/cmd/dist/README#17) says, "Releases ≥ Go 1.x are very likely to work as well."

**Comment by 4a6f656c** [comment]
> @bcmills Can we upgrade GOROOT_BOOTSTRAP to Go 1.20.7 and buildlet to Go 1.21? This fixed a LD/SC asm code generation issue(#61295)I think this might related.

@mengzhuo is this still a SiFive HiFive Unmatched machine?

**Comment by mengzhuo** [comment]
> > @bcmills Can we upgrade GOROOT_BOOTSTRAP to Go 1.20.7 and buildlet to Go 1.21? This fixed a LD/SC asm code generation issue(#61295)I think this might related.
> 
> @mengzhuo is this still a SiFive HiFive Unmatched machine?

There is one Unmatched(unmatched025.farm.rvlab.org), I want to improve the test time of trybot by adding more SBC (visionfive2 or Licheepi4a) in the farm. Unfortunately I can't get more new Unmatched since it's discontinued.

I've looked all failed logs, all failed on Licheepi4a, I had offlined this SBC.

My initial guess is Licheepi4a using different architecture (Thead Xuantie 9XX) instead of Visionfive2/Unmatched (Sifive u74). They might be different in the OoO and atomic.
